pkg/local_object_storage/metabase/graveyard.go

```diff
@@ -299,15 +298,23 @@ func (db *DB) GetGarbage(limit int) ([]oid.Address, []cid.ID, error) {
 		// also be deleted as a part of non-existing
 		// container so no need to handle it twice
 
-		bkt := tx.Bucket(garbageContainersBucketName)
-		c := bkt.Cursor()
-
-		for k, _ := c.First(); k != nil; k, _ = c.Next() {
-			err := cidBuff.Decode(k)
-			if err != nil {
-				return fmt.Errorf("parsing raw CID: %w", err)
+		var inhumedCnrs []cid.ID
+		err := tx.ForEach(func(name []byte, b *bbolt.Bucket) error {
+			if name[0] == metadataPrefix && containerMarkedGC(b.Cursor()) {
+				var cnr cid.ID
+				cidRaw, prefix := parseContainerIDWithPrefix(&cnr, name)
+				if cidRaw == nil || prefix != metadataPrefix {
+					return nil
+				}
+				inhumedCnrs = append(inhumedCnrs, cnr)
 			}
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("scanning inhumed containers: %w", err)
+		}
 
+		for _, cidBuff := range inhumedCnrs {
```

Comment:
```suggestion
		for _, cnr := range inhumedCnrs {
```
it's not buffer anymore. Although i understand diff will grow a bit